### DISCUSSION TITLE
Randomization for different conditions

### DISCUSSION THREAD

**Discussion by SaoirseConnorDesai** [jspsych/jsPsych#941] [Q&A]
I trying to set-up an experiment with some pretty complex randomization procedures that vary by condition, but am having trouble figuring out a sensible way to do this. 

Here's the overview of the experiment: 

I have a set of five 'articles' that participants should read in random order (four opposing, one supporting). In three of the conditions, they should see all five articles in random order. In the fourth condition, they need to always see one of the articles and then one of the remaining four articles which has been randomly selected (or counterbalanced across participants if that's easier to do). 

I also need to randomise the name of the source within each article. In two of the conditions, each article should have a randomly selected source from a set of five sources. In the other condition that sees five articles, they should see the same source in four of the articles and a different source in one specific article. Then in the final condition, where they see only two articles, they should see a randomly selected source in each article. 

The final randomization procedure is the background colour of the source in each article. In the conditions where participants see different sources, the background colour of the source needs to be different but in the condition in which they see the same source in four different articles, I need to use the same background colour for those four articles, and a different one for the final article. For the two article condition, the background colour just needs to be different for each article. 

I realise this is quite a lot but at the moment I don't even know where to start so was hoping someone might have some ideas to point me in the right direction. 

Currently, all I've managed is to get the order to randomise for all five articles:

```
var essayOrder1 = jsPsych.randomization.shuffle([essay1, essay2, essay3, essay4, essay5]);

        jsPsych.init({
            timeline: essayOrder1,
            // on_trial_finish: function (data) { saveData(filename, jsPsych.data.get().csv()) },  // use for UNSW server - uncomment to locally save data file as csv
            on_finish: function () {
                endExperiment(jsPsych.data.get().csv(), function () { document.write('<div id="endscreen" class="endscreen" style="width:1000px"><div class="endscreen" style="text-align:center; border:0px solid; padding:10px; font-size:120%; width:800px; float:right"><p><br><br><br>All done!<br><br>Your completion code is <span id="turkcode" style="font-weight:bold;font-size:130%">' + turkcode + '</span>. To receive payment for the HIT, return to the Amazon Mechanical Turk page and enter this code. Please contact us if something goes wrong and we\'ll fix it as quickly as possible.</p></div></div>') })
            }
        });
    };

timeline.push(essayOrder1);
```

**Comment by SaoirseConnorDesai** [comment]
I've found a long-winded way of doing this by using if / else if statements for each different condition. 

The issue that I'm having now is that because I specify the timeline as essaryOrder1 in the jsPsych.init I'm not able to push anything else to the timeline before this.

**Reply by jodeleeuw** [reply]
Can you create a timeline array, and then add `essayOrder1` to that timeline?

```js
var timeline = [];

timeline = timeline.concat(essayOrder1); // will add all 5 essay trials to the timeline

jsPsych.init({
	timeline: timeline
	...
})
```
